### DISCUSSION TITLE
Need help understanding the Codebase

### DISCUSSION THREAD

**Discussion by kingbass01** [oppia/oppia-android#4817] [Q&A (Navigating the codebase)] [answered]
I have been trying to understand the codebase and how it works but I don't think I am making any progress and everytime I open a new file it makes me more confused. I am getting errors in many files without even touching them but then also the app builds succesfully. The wiki also gives overview but can't help.
I tried these examples also : 
[Example 1](https://github.com/oppia/oppia-android/wiki/Overview-of-the-Oppia-Android-codebase-and-architecture#example-1)
[Example 2](https://github.com/oppia/oppia-android/wiki/Overview-of-the-Oppia-Android-codebase-and-architecture#example-2)
But they also seem outdated because there is no info tab in the app when I ran it
<img src="https://user-images.githubusercontent.com/85404940/210128067-aa3a3cda-a316-439d-94ce-b2d910eec170.png" width="200">

I was also not assigned any mentor after filling CLA and feedback form to whom I can ask my doubts.
Any help regarding how to understand and start could help because I really want to contribute.
Thanks.

**Comment by MohitGupta121** [comment, marked as answer]
Thanks for your interset in contributing to oppia.

@kingbass01 Which Android Studio version you are using?
Recommended version for oppia-android is - **Android Studio Bumblebee (Patch 3)** 
Please try this and let me know if you getting any errors.

Yes actually that [wiki](https://github.com/oppia/oppia-android/wiki/Overview-of-the-Oppia-Android-codebase-and-architecture#directory-structure) is just an example how to navigate to code and understanding the structure now in new UI there is only two tabs you get get idea from [here](https://github.com/oppia/oppia-android/wiki/Overview-of-the-Oppia-Android-codebase-and-architecture#directory-structure).

Can you please check you recived any CLA reply email for contributor-survey form?

Please fill contributor-survey form that you get in CLA reply email after that you get mentor.

Hope this will helps you.
If you have any query feel free to ask on [github-discussions](https://github.com/oppia/oppia-android/discussions).

If this helps you please mark this as answer so that it's help others in community.

**Reply by kingbass01** [reply]
Thank you for getting back to me.
Yes on using **Android Studio Bumblebee (Patch 3)** all errors got fixed and felt much lighter and smoother. 

I will try reading the wiki again and ask if I have any doubts.

And yes I received a reply after filling out the CLA form and filled the contributor-survey form then only but I didn't receive any response after that.
But I doubt that this question references only to the web repository of oppia or android also because my reply to this was **No** because I thought it is asking about the web repository only. Can you please confirm?
![image](https://user-images.githubusercontent.com/85404940/210194484-09d83b1b-5e41-4967-8a67-5588acbd9a90.png)

Thanks.

**Reply by MohitGupta121** [reply]
Hey @kingbass01 actually you have to select 'Yes' if you are interested in contributing to Oppia codebase (android or web) is decided and totally on your interest and according to your contributor-survery report.

**Reply by seanlip** [reply]
Yup, just confirming that this is referring to both codebases. I've updated the form to clarify, since it was originally written at a time when we had just one codebase. Thanks for bringing this up!

**Reply by kingbass01** [reply]
Ok, I edited my response and resubmitted the form.
Thanks for the help.
